community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/planner/logical/CardinalityCostModel.scala

```diff
@@ -41,11 +43,15 @@ object CardinalityCostModel extends CostModel {
     => 1.0
 
     // Filtering on labels and properties
-    case Selection(predicates, _) if predicates.exists {
-      case _: Property | _: HasLabels => true
-      case _ => false
-    }
-    => 1.0
+    case Selection(predicates, _) =>
+      val noOfStoreAccesses = predicates.treeCount {
+        case _: Property | _: HasLabels => true
+        case _ => false
+      }
+      if (noOfStoreAccesses > 0)
+        CostPerRow(noOfStoreAccesses)
+      else
+        DEFAULT_COST_PER_ROW
```

Comment:
This is a reduction of a factor 10 for the default case. Perhaps it would be sensible to also take into account how large the `predicates` sequence is?